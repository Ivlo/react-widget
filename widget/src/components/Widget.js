/* eslint-disable react/button-has-type */
import React, { useState, useContext, useCallback } from "react";
import WidgetContext from "../WidgetContext";
import { setEvent } from "../services/analyticsEvents";

import Select from "./Select";
import Modal from "./Modal";
import Loader from "./Loader";
import FinancingCost from "./FinancingCost";

const Widget = ({ financingData }) => {
  const [isModalVisible, setIsModalVisible] = useState(false);
  const { setInstalmentFee } = useContext(WidgetContext);

  function sendAnalyticEvent(selectedInstalment) {
    setEvent({
      context: "checkoutWidget",
      type: "simulatorInstalmentChanged",
      selectedInstalment,
    });
  }

  const handleSelectCallback = useCallback((event) => {
    const selectedInstalment = event.target.value;
    if (selectedInstalment) {
      setIsModalVisible(true);
      setInstalmentFee(selectedInstalment);
      sendAnalyticEvent(selectedInstalment);
    }
  }, []);

  function onClickCloseModal() {
    setIsModalVisible(false);
  }

  return (
    <>
      {financingData ? (
        <>
          <p>Pagalo en</p>
          <p>
            <a href="#">m√°s info</a>
          </p>
          <Select
            handleChange={handleSelectCallback}
            optionsData={financingData}
            optionDefault="Select a element"
            optionsText="cuotas de"
          />
          {isModalVisible && (
            <Modal onClickClose={onClickCloseModal}>
              <FinancingCost />
            </Modal>
          )}
        </>
      ) : (
        <Loader text="Loading..." />
      )}
    </>
  );
};

export default Widget;
