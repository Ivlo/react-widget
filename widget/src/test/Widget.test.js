import React from "react";
import { shallow } from "enzyme";
import Widget from "../components/Widget";

const wrapper = shallow(<Widget />);

describe("Widget", () => {
  it("should render the Widget Component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
