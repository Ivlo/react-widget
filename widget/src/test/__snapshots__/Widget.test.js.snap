// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget should render the Widget Component correctly 1`] = `<Fragment />`;
